src/SessionSetup/RetrieveInstancesWidget.cpp

```diff
@@ -160,6 +172,14 @@ void RetrieveInstancesWidget::OnInitialLoadingReturnedSuccess(
     }
   }
 
+  // This projectComboBox currentIndexChanged connection is only set up after the
+  // combo box is filled. If it would be set up earlier (for example in the
+  // constructor), then the first selection that just happened (setCurrentIndex) would
+  // already trigger a OnProjectComboCurrentIndexChanged which then might trigger
+  // another load that is unneccessary.
+  QObject::connect(ui_->projectComboBox, QOverload<int>::of(&QComboBox::currentIndexChanged), this,
+                   &RetrieveInstancesWidget::OnProjectComboBoxCurrentIndexChanged);
```

Comment:
You should use the Qt::UniqueConnection flag (https://doc.qt.io/qt-5/qt.html#ConnectionType-enum) here to ensure the slot is not called multiple times. (Or is there a disconnect call somewhere that I missed?)

Reply:
I moved the connect into the constructor and used blockSignal where appropriate